src/main.py

```diff
@@ -48,42 +57,109 @@ async def root():
 # Could look at 'get_forecast_metadata' to see when this forecast is made
 
 # get_sites: Clients get the site id that are available to them
-@app.get("/sites/site_list")
-async def get_sites(pv_sites: PV_Sites):
-    return {"pv_sites": PV_Sites}
+@app.get("/sites/site_list", response_model=PV_Sites)
+async def get_sites():
+    # simple 2. (fake just return a list of one site using 'fake_site_uuid'
+    pv_site = PV_Site_Metadata(
+        uuid=fake_site_uuid, site_name="fake site name", latitude=50, longitude=0, capacity_kw=1
+    )
+    pv_site_list = PV_Sites(
+        site_list=[pv_site],
+    )
+
+    return pv_site_list
 
 
 # # post_pv_actual: sends data to us, and we save to database
 @app.post("/sites/pv_actual/{site_id}")
-async def post_pv_actual(one_pv_actual: One_PV_Actual):
-    return {"one_pv_actual": One_PV_Actual}
+async def post_pv_actual(
+    site_uuid: str,
+    pv_actual: One_PV_Actual,
+):
+    # simple 4. (fake = just return what is put in)
+
+    print(f"Got {pv_actual.dict()} for site {site_uuid}")
+    print("Not doing anything with it (yet!)")
 
 
 # put_site_info: client can update a site
-@app.put("/sites/pv_actual/{site_id}/info")
-async def put_site_info(site_metadata: PV_Site_Metadata):
-    return {"site_metadata": PV_Site_Metadata}
+@app.put("/sites/pv_actual/{site_id}/info", response_model=PV_Site_Metadata)
+async def put_site_info(
```

Comment:
see above, we have passed an object in. 
So for this route we should pass a `PV_Site_Metadata` metadata in. We can then return nothing. In the real thing this will update the database

Reply:
thanks for all the comments @peterdudfield, I'll have a look at the code today and tidy it up.